src/com/amazon/ionelement/api/AnyElement.kt

```diff
@@ -0,0 +1,335 @@
+ * val value: String = e.textValue
+ * // throws if (e is null) or (not a STRING or SYMBOL).
+ * ```
+ *
+ * Narrowing Function Examples:
+ *
+ * ```
+ * val e: AnyElement = loadSingleElement("1")
+ * val n: IntElement = e.asInt()
+ * // e.asInt() throws if e is null or not an INT
+ * ```
+ *
+ * ```
+ * val e: AnyElement = loadSingleElement("[1, 2]")
+ * val l: ListElement = e.asList()
+ * // e.asList() throws if e is null or not a list
+ *
+ * val values: List<IntElement> = l.values.map { it.asInt() }
+ * // l.values.map { it.asInt() } throws if: any child element in l is null or not an int
+ * ```
+ *
+ * ```
+ * val e: AnyElement = loadSingleElement("some_symbol")
+ * val t: String = e.textValue
+ * // throws if (e is null) or (not a STRING or SYMBOL).
+ * ```
+ *
+ * #### Deciding which accessor function to use
+ *
+ * **Note:  for the sake of brevity, the following section omits the nullable narrowing functions (`as*OrNull`) and
+ * nullable value accessors (`*OrNull`).  These should be used whenever an Ion-null value is allowed.
+ *
+ * The table below shows which accessor functions can be used for each [ElementType].
+ *
+ * | [ElementType]    | Value Accessors                                 | Narrowing Functions                    |
+ * |------------------------------|------------------------------------------------------------------------------|
+ * | [NULL]           | (any with `OrNull` suffix)                      | (any with `OrNull` suffix)             |
+ * | [BOOL]           | [booleanValue]                                  | [asBoolean]                            |
+ * | [INT]            | [longValue], [bigIntegerValue]                  | [asInt]                                |
+ * | [STRING]         | [textValue], [stringValue]                      | [asText], [asString]                   |
+ * | [SYMBOL]         | [textValue], [symbolValue]                      | [asText], [asSymbol]                   |
+ * | [DECIMAL]        | [decimalValue]                                  | [asDecimal]                            |
+ * | [TIMESTAMP]      | [timestampValue]                                | [asTimestamp]                          |
+ * | [CLOB]           | [bytesValue], [clobValue]                       | [asLob], [asClob]                      |
+ * | [BLOB]           | [bytesValue], [blobValue]                       | [asLob], [asBlob]                      |
+ * | [LIST]           | [containerValues], [seqValues], [listValues]    | [asContainer], [asSeq], [asList]       |
+ * | [SEXP]           | [containerValues], [seqValues], [sexpValues]    | [asContainer], [asSeq], [asSexp]       |
+ * | [STRUCT]         | [containerValues], [structFields]               | [asContainer], [asStruct]              |
+ *
+ * Notes:
+ * - The value returned from [containerValues] when the [type] is [STRUCT] is the values within the struct without
+ * their field names.
+ */
+interface AnyElement : IonElement {
```

Comment:
Shall we just call `IonElement` as `Element`?  Is the Ion prefix buying us anything?

Reply:
A little less ambiguity perhaps.  `Element` is commonly used by many types in the accessible in my IDE's code completion classpath.